### ISSUE TITLE
[cdk-pipelines] Reuse CodeBuild projects between assets

### ISSUE THREAD

**Issue by rix0rrr** [aws/aws-cdk#9629]
A CodeBuild project per asset is a little much, especially since we could reuse 1 CodeBuild project for file assets and 1 CodeBuild project for docker assets, and invoke them differently for every asset type.

---

This is a :rocket: Feature Request

**Comment by adelsz** [comment]
Good point, having a huge list of CodeBuild projects for a not-so-big CDK stack does look weird.
